index.js

```diff
@@ -59,21 +60,39 @@ async function getCoreIndex () {
   }
 }
 
-function getVulnerabilityList (currentVersion, data) {
+function getVulnerabilityList (currentVersion, data, systemEnvironment) {
   const list = []
   for (const key in data) {
     const vuln = data[key]
     if (
-      satisfies(currentVersion, vuln.vulnerable) &&
-      !satisfies(currentVersion, vuln.patched)
+      (
+        satisfies(currentVersion, vuln.vulnerable) &&
+        !satisfies(currentVersion, vuln.patched)
+      ) && (
+        (!systemEnvironment || !Array.isArray(vuln.affectedEnvironments)) ||
+        vuln.affectedEnvironments.includes(systemEnvironment) ||
+        vuln.affectedEnvironments.includes('all')
```

Comment:
I'd split it into a new function: `affectCurrentSystemEnvironment` or something more descriptive

Reply:
Added new function `isSystemAffected` https://github.com/RafaelGSS/is-my-node-vulnerable/pull/12/commits/2a277aef3d3c19e661433cb2e7def4eb9b636a7c